kinto_client/__init__.py

```diff
@@ -117,265 +110,113 @@ def request(self, method, url, data=None, permissions=None, **kwargs):
-            Permissions to be used when creating a bucket.
-        :param server_url:
-            The URL of the server to use.
-        :param auth:
-            A requests authentication policy object.
-        :param session:
-            An optional session object to use, rather than creating a new one.
-        :param create:
-            Defines if the bucket should be created. (defaults to False)
-        :param load:
-            Defines if bucket data should be loaded or not (defaults to True)
-        """
+    def __init__(self, server_url=None, session=None, auth=None,
+                 bucket=None, collection=None):
         self.session = create_session(server_url, auth, session)
-        self.name = utils.slugify(name)
-        self.uri = '/buckets/%s' % self.name
-        self.permissions = Permissions(object='bucket')
-        self.data = None
-
-        if load:
-            # XXX put this logic in a separate method.
-            method = 'put' if create and name != 'default' else 'get'
-
-            # In the case of a creation, check if permissions have been passed.
-            kwargs = {}
-            if method == 'put':
-                kwargs['permissions'] = permissions
-
-            try:
-                info, _ = self.session.request(method, self.uri, **kwargs)
-            except KintoException as e:
-                if method == 'get' and e.response.status_code == 403:
-                    exception = BucketNotFound(name)
-                    exception.response = e.response
-                    exception.request = e.request
-                    raise exception
-                else:
-                    raise
-
-            self.data = info['data']
-            self.permissions = Permissions(
-                object='bucket',
-                permissions=info['permissions'])
-
-    def _get_collection_uri(self, collection_id):
-        return '%s/collections/%s' % (self.uri, collection_id)
-
-    def get_collection(self, name, **kwargs):
-        return Collection(name, bucket=self, session=self.session, **kwargs)
-
-    def list_collections(self):
-        uri = "%s/%s" % (self.uri, 'collections')
-        resp, _ = self.session.request('get', uri)
-
-        return [collection[ID_FIELD] for collection in resp['data']]
-
-    def create_collection(self, name, permissions=None):
-        return Collection(name, bucket=self, session=self.session,
-                          create=True, permissions=permissions)
-
-    def delete_collection(self, name):
-        uri = self._get_collection_uri(name)
-        resp, _ = self.session.request('delete', uri)
+        self._bucket_name = bucket
+        self._collection_name = collection
+        self.endpoints = Endpoints()
+
+    def _get_endpoint(self, name, bucket=None, collection=None, id=None):
+        kwargs = {
+            'bucket': bucket or self._bucket_name,
+            'collection': collection or self._collection_name,
+            'id': id
```

Comment:
for this specific case, I think it's okay to shadow the builtin "id" here. If we need it somewhere, we'll find our way around for sure. (I find `record_id` to be quite verbose). compare

```python
client.update_record(id=1324, data={'foo': 'bar'})
```
with
```python
client.update_record(record_id=1234, data={'foo': 'bar'})
```